magma/t.py

```diff
@@ -142,54 +142,50 @@ def flip(cls):
             return cls
 
 
-def dispatch_magma_protocol(func):
-    @functools.wraps(func)
-    def wrapper(T):
-        if issubclass(T, MagmaProtocol):
-            return T._from_magma_(func(T._to_magma_()))
-        return func(T)
-    return wrapper
-
-
-@dispatch_magma_protocol
 def In(T):
     return T.qualify(Direction.In)
```

Comment:
Don't we still need to dispatch to the protocol? E.g.  `In(MagmaProtocol_t)` needs to dispatch to  `MagmaProtocol_t._qualify_magma_(In)`

Reply:
Oh wait that's in `MagmaProtocol.qualify`